src/Chainweb/Sync/WebBlockHeaderStore.hs

```diff
@@ -313,9 +317,11 @@ getBlockHeaderInternal headerStore payloadStore candidateHeaderCas candidatePayl
         -- prerequesite in the memo-table it is awaited, otherwise a new job is
         -- created.
         --
-        let queryAdjacentParent p = Concurrently $ void $ do
-                logg Debug $ taskMsg k $ "getBlockHeaderInternal.getPrerequisteHeader (adjacent) for " <> sshow h <> ": " <> sshow p
-                getBlockHeaderInternal headerStore payloadStore candidateHeaderCas candidatePayloadCas priority maybeOrigin' p
+        let isGenesisParentHash p = _chainValueValue p == genesisParentBlockHash v p
+            queryAdjacentParent p = Concurrently $ unless (isGenesisParentHash p) $ void $ do
```

Comment:
Yes. It would require careful refactoring of the code around line 345. 

The overhead of creating a thread in the Haskell runtime is really minimal and there is some chance that the guard can be floated out during optimization, so, I think, it's not worth refactoring the code at this point.